deps/wazuh_testing/wazuh_testing/vulnerability_detector.py

```diff
@@ -25,6 +26,45 @@ def callback_detect_vulnerability_scan_sleeping(line):
     return match.group(1) if match is not None else ""
 
 
+def callback_detect_update_from_year_result(line):
+    """
+    Check if the selected provider was added with the correct year to update sinze
+    Also, check if modulesd have failed due to a invalid year selected as update_from_year value
+
+    returns a triple:
+    ("result", "provider", "update_from_value")
+    result can be: "invalid_year", "provider_added"
+    """
+    invalid_year_msg = r'.*wazuh-modulesd.*wmodules-vuln-detector.*ERROR: Invalid content for \'update_from_year\' option at module \'vulnerability-detector\''
+    invalid_year_match = re.match(invalid_year_msg, line)
+
+    if invalid_year_match is not None:
+        return "invalid_year", None, None
+
+    provider_added_msg = r'.*wazuh-modulesd.*Added (.*) feed.*Update since: (.*)\.'
+    provider_added_match = re.match(provider_added_msg, line)
+
+    if provider_added_match is not None:
+        return "provider_added", provider_added_match.group(1), provider_added_match.group(2)
+
+    return False
```

Comment:
That wouldn't work, the way this callback is used is: match each line of the log until return something different from `False`. I mean, you need to return false if nothing has matched so the log file is completely read.

Reply:
The problem of this is sometimes the function may return different objects depending on the result of the callback. For me, this doesn't keep consistency but since the `FileMonitor` is thought to work using this...